### ISSUE TITLE
Add prediction-level feature importances?

### ISSUE THREAD

**Issue by Vincent-Maladiere** [skrub-data/skrub#1144]
### Problem Description

For data scientists in the industry, explainability is crucial to build trust with other stakeholders. Marginal explainability with, e.g., `permutation_importance` helps, but stakeholders often ask for personalised, per-prediction feature importances, which help understand what stands out in a test sample for the model.

The two most popular librairies for this are LIME and SHAP, both having theoretical and practical pros and cons. Additionally, LIME is not maintained anymore, although fairly simple conceptually (fitting Ridge models while varying sample weights).

This raises several questions:

1. Would having an easy way to get marginal and prediction-level feature importance make sense in the recipe or as a standalone helper in skrub?
2. If we don't want extra dependancies, should we implement a simple version of prediction-level feat imp in skrub? @glemaitre mentioned one SLEP with a similar objective, IIUC
3. We should probably do some benchmarking and litterature reading to choose a method which can be interpreted and explained in plain english, with clear documentation and guidelines on what can or cannot be said.

WDYT?

### Feature Description

.

### Alternative Solutions

.

### Additional Context

.

**Comment by GaelVaroquaux** [comment]
I'm not sure why this belongs in skrub? It seems a problem/feature fairly generic to ML

**Comment by Vincent-Maladiere** [comment]
I agree that this raises the question of the scope, too. Skrub and the recipe already ease HP tuning, so it could be valuable for users to also facilitate model evaluation IMHO, with pragmatic guidelines. I'd also be happy to have it somewhere else, though.

**Comment by Vincent-Maladiere** [comment]
Won't be addressed in skrub

**Comment by GaelVaroquaux** [comment]
> Won't be addressed in skrub

I actually think that it would be very interesting to considering add an inspection step (typically feature importances) in the middle of a pipeline built by expression. A bit like the "choose_from" magic.
